src/cachemachine/types.py

```diff
@@ -102,6 +102,22 @@ def dump(self) -> List[Dict[str, str]]:
         return [asdict(i) for i in self]
 
 
+@dataclass
+class DesiredImageList:
+    """Container for the results of a desired_image call.
+
+    Contains two lists, one the images to pull, and the other
+    containing all the other images that could be pulled, but
+    won't be.
+    """
+
+    """Images that should be pulled."""
+    desired_images: DockerImageList
+
+    """Images that should end up in a sublist, but not be pulled."""
+    all_images: DockerImageList
```

Comment:
If it really is XORed then maybe call it undesired_images?  If it is all images, then the comment should be something like "All images, whether or not they should be pulled."